### PR TITLE
Feat/copy database

### PR DESCRIPTION
**Pull request by tonsV2**
Upon implementing this feature it became apparent that create/upload needed to be merged. Therefore this PR also includes a new "upload" endpoint implementing that.

An update to the path under which a database was stored in S3 is also part of this PR.
Previously databases was stored as :group/:name/:name.:filetype.
Now the path is just :group/:name (name is the original name of the file which the user uploaded). Please note that this can be changed via the "update" endpoint.

### CODE REVIEW

File: pkg/storage/s3.go [hunk 1]
```diff
@@ -23,6 +24,22 @@ func NewS3Client() S3Client {
...
+	}
+
+	_, err = awsS3Client.CopyObject(context.TODO(), &s3.CopyObjectInput{
+		Bucket:     aws.String(bucket),
+		CopySource: aws.String(bucket + "/" + source),
+		Key:        aws.String(destination),
+		ACL:        types.ObjectCannedACLPrivate,
+	})
+
+	return err
```
[the remark this answers is not in the document]
Comment (tonsV2) [reply pkg/storage/s3.go:40]: Great idea! I've implemented it for the other methods as well

File: pkg/storage/s3.go [hunk 2]
```diff
@@ -23,6 +24,22 @@ func NewS3Client() S3Client {
 	return s3Client{}
 }
 
+func (s s3Client) Copy(bucket string, source string, destination string) error {
+	awsS3Client, err := s.getS3Client()
```
[the remark this answers is not in the document]
Comment (tonsV2) [reply pkg/storage/s3.go:28]: Would you define the new method as below?

    func NewS3Client() (S3Client, error)

**Review by teleivo — APPROVED** [review]

File: pkg/database/handler.go [hunk 3]
```diff
@@ -52,25 +130,45 @@ type CreateDatabaseRequest struct {
 //   401: Error
 //   403: Error
 //   415: Error
-func (h Handler) Create(c *gin.Context) {
-	var request CreateDatabaseRequest
+func (h Handler) Copy(c *gin.Context) {
+	idParam := c.Param("id")
+	id, err := strconv.ParseUint(idParam, 10, 64)
+	if err != nil {
+		badRequest := apperror.NewBadRequest("error parsing id")
```
[the remark this answers is not in the document]
Comment (teleivo) [reply pkg/database/handler.go:137]: postponed

File: pkg/database/service.go [hunk 4]
```diff
@@ -50,6 +51,33 @@ func (s service) Create(d *model.Database) error {
 	return s.repository.Create(d)
 }
 
+func (s service) Copy(id uint, d *model.Database, group *models.Group) error {
+	source, err := s.FindById(id)
```
[the remark this answers is not in the document]
Comment (teleivo) [reply pkg/database/service.go:55]: postponed

**Pull request merged**